internal/super/all_help.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2024 Flow Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package super
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var ansiRegex = regexp.MustCompile(`\x1B\[[0-9;]*[a-zA-Z]`)
+
+var AllHelp = &cobra.Command{
+	Use:     "all-help",
+	Short:   "Outputs help for all the CLI commands",
+	Example: "flow cheatsheet",
+	Run: func(
+		c *cobra.Command,
+		args []string,
+	) {
+		root := c.Root()
+		r, err := generateCS(root)
+		if err != nil {
+			fmt.Printf("Error generating cheatsheet: %s", root.Name())
+		}
+		fmt.Println(r)
+	},
+}
+
+func generateCS(c *cobra.Command) (string, error) {
+	cmd := c.Root()
+	var helpTexts strings.Builder
+
+	helpTexts.WriteString("```")
+	// Generate the help texts
+	generateCommandHelpTexts(cmd, &helpTexts)
+	helpTexts.WriteString("```")
+
+	return helpTexts.String(), nil
+
+}
+
+// Recursive function to traverse all commands and subcommands,
+// capturing the help text for each.
+func generateCommandHelpTexts(cmd *cobra.Command, result *strings.Builder) {
+	removeGlobalFlags := true
+	// version is last command show global flags
+	if cmd.Name() == "version" {
```

Comment:
Checking the string name here is prone to break at some point. Can you check for the last item instead please?